packages/modals/src/styled/StyledModal.ts

```diff
@@ -47,9 +52,8 @@ const sizeStyles = (props: IStyledModalProps & ThemeProps<DefaultTheme>) => {
   const largeWidth = 800;
 
   return css`
-    width: ${props.isLarge ? `${largeWidth}px` : `${defaultWidth}px`};
-    @media (max-width: ${props.isLarge ? `${largeWidth - 1}px` : props.theme.breakpoints.md}) {
-      ${props.theme.rtl ? 'right' : 'left'}: ${props.theme.space.base * 6}px;
+    ${mediaQuery('up', props.isLarge ? 'md' : 'sm', props.theme)} {
+      width: ${props.isLarge ? `${largeWidth}px` : `${defaultWidth}px`};
```

Comment:
In order to make this perfectly smooth responsive, this line should be

```suggestion
      width: ${props.isLarge ? props.theme.breakpoints.md : props.theme.breakpoints.sm};
```

(the `defaultWidth` and `largeWidth` variables can be removed)